core/exec/src/mill/exec/Execution.scala

```diff
@@ -147,8 +148,14 @@ private[mill] case class Execution(
 
               val verboseKeySuffix = s"/${terminals0.size}"
               logger.setPromptHeaderPrefix(s"$countMsg$verboseKeySuffix")
```

Comment:
It seems like all your problems with the `, failed` suffix disappearing is because of this line. If you update this line to contain the suffix, you won't need to put the weird code in the `failed.get()` branch to immediately overwrite it